orchagent/fdborch.cpp

```diff
@@ -482,26 +700,73 @@ bool FdbOrch::addFdbEntry(const FdbEntry& entry, const string& port_name, const
     attr.value.s32 = SAI_PACKET_ACTION_FORWARD;
     attrs.push_back(attr);
 
-    if (m_entries.count(entry) != 0) // we already have such entries
+    string key = "Vlan" + to_string(vlan.m_vlan_info.vlan_id) + ":" + entry.mac.to_string();
+    
+    if(macUpdate)
     {
-        removeFdbEntry(entry);
+        /* delete and re-add fdb entry instead of update,
+         * as entry may age out in HW/ASIC_DB before
+         * update, causing the update request to fail.
+         */
+        SWSS_LOG_NOTICE("MAC-Update FDB %s in %s on from-%s:to-%s from-%s:to-%s", entry.mac.to_string().c_str(), vlan.m_alias.c_str(), oldPort.m_alias.c_str(), port_name.c_str(), oldType.c_str(), type.c_str());
+        status = sai_fdb_api->remove_fdb_entry(&fdb_entry);
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_INFO("FdbOrch RemoveFDBEntry: Failed to remove FDB entry. mac=%s, bv_id=0x%lx",
+                           entry.mac.to_string().c_str(), entry.bv_id);
+        }
+        else
+        {
+            oldPort.m_fdb_count--;
+            m_portsOrch->setPort(oldPort.m_alias, oldPort);
+            if (oldPort.m_bridge_port_id == port.m_bridge_port_id)
```

Comment:
shouldn't this to be "if (oldPort.m_bridge_port_id != port.m_bridge_port_id)"

Reply:
we retrieved both oldport and "port". refcount is decremented for oldport. if "port" is same as old port. we need to update "port" refcount also, otherwise when we increment the refcount for "port", it will be incorrect. We need not setPort with "port", but it doesn't hurt to set.